frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -10,74 +9,94 @@ import {
 import './_add-capacity-modal.scss';
+import { createDeviceSet, DeviceSet } from '../../ocs-install/ocs-request-data';
+import { cephCapacityResource } from '../../../constants/resources';
 
 const getProvisionedCapacity = (value: number) => (value % 1 ? (value * 3).toFixed(2) : value * 3);
 
+const getCurrentDeviceSet = (deviceSets: DeviceSet[], selectedSCName: string): number =>
+  deviceSets.findIndex((ds) => ds.dataPVCTemplate.spec.storageClassName === selectedSCName);
+
 export const AddCapacityModal = (props: AddCapacityModalProps) => {
   const { ocsConfig, close, cancel } = props;
+  const deviceSets: DeviceSet[] = ocsConfig?.spec?.storageDeviceSets || [];
 
-  const [response, loadError, loading] = usePrometheusPoll({
-    endpoint: PrometheusEndpoint.QUERY,
-    namespace: CEPH_STORAGE_NAMESPACE,
-    query: CAPACITY_USAGE_QUERIES[StorageDashboardQuery.CEPH_CAPACITY_USED],
-  });
+  const [response, loadError, loading] = usePrometheusPoll(cephCapacityResource);
   const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
+  /* TBD(Afreen): Show installation storage class as preselected 
+                  Change state metadata
+  */
   const [inProgress, setProgress] = React.useState(false);
   const [errorMessage, setError] = React.useState('');
 
-  const osdSizeWithUnit = _.get(
-    ocsConfig,
-    'spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage',
-  );
-  const presentCount = _.get(ocsConfig, 'spec.storageDeviceSets[0].count');
-  const capacity = _.get(response, 'data.result[0].value[1]');
-
-  const osdSizeWithoutUnit = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
+  const cephCapacity: string = response?.data?.result?.[0]?.value[1];
+  const osdSizeWithUnit = getRequestedPVCSize(deviceSets[0].dataPVCTemplate);
+  const osdSizeWithoutUnit: number = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
   const provisionedCapacity = getProvisionedCapacity(osdSizeWithoutUnit);
+  const isNoProvionerSC: boolean = storageClass?.provisioner === NO_PROVISIONER;
+  const selectedSCName: string = getName(storageClass);
+  const deviceSetIndex: number = getCurrentDeviceSet(deviceSets, selectedSCName);
+  const { replica } = deviceSets[deviceSetIndex];
 
   let currentCapacity: React.ReactNode;
 
   if (loading) {
     currentCapacity = (
       <div className="skeleton-text ceph-add-capacity__current-capacity--loading" />
     );
-  } else if (loadError || !capacity || !presentCount || !osdSizeWithoutUnit) {
+  } else if (loadError || !cephCapacity || !osdSizeWithoutUnit || deviceSetIndex === -1) {
     currentCapacity = <div className="text-muted">Not available</div>;
   } else {
     currentCapacity = (
       <div className="text-muted">
-        <strong>{`${humanizeBinaryBytes(capacity / 3).string} / ${presentCount *
-          osdSizeWithoutUnit} TiB`}</strong>
+        <strong>{`${humanizeBinaryBytes(+cephCapacity / replica).string} / ${deviceSets[
```

Comment:
```suggestion
        <strong>{`${humanizeBinaryBytes(Number(cephCapacity) / replica).string} / ${deviceSets[
```
Or you can let it be as it is `cephCapacity / replica`. It converts automatically.

Reply:
Yeah it was doing automatically back then, but its not right to do that.
Operations or usage might change later, so adding type and converting seems fine.

Reply:
If you want type coercion than let it handle by `/` else if you want to make it explicit use `Number` but using `+` is exactly the same as letting `/` handle it.

Reply:
You are right. Makes sense